lib/board.rb

```diff
@@ -29,30 +29,32 @@ def valid_coordinate?(coordinate)
   end
 
   def valid_placement?(ship, coordinates)
-    if ship.length == coordinates.length &&
-      coordinates.all? {|coordinate| valid_coordinate?(coordinate)} &&
-      coordinates.all? {|coordinate| @cells[coordinate].empty?}
-      consecutive?(coordinates)
+    if check_validation(ship, coordinates)
+      consecutive?(coordinates.sort)
     else
       false
     end
   end
 
-  def set_letters(coordinates)
+  def check_validation(ship, coordinates)
+    ship.length == coordinates.length &&
+      coordinates.all? {|coordinate| valid_coordinate?(coordinate)} &&
+      coordinates.all? {|coordinate| @cells[coordinate].empty?}
+  end
+
+  def letters(coordinates)
     (coordinates[0][0]..coordinates[-1][0]).to_a
   end
 
-  def set_numbers(coordinates)
+  def numbers(coordinates)
     (coordinates[0][1]..coordinates[-1][1]).to_a
   end
```

Comment:
The letters and numbers methods are going to work so well if we try iteration 4. These lines of code really make me happy.

Reply:
Yay! Glad you like it!